src/common/snippets/src/lowered/pass/insert_loops.cpp

```diff
@@ -80,6 +81,17 @@ void InsertLoops::insertion(LinearIR& linear_ir, const LinearIR::LoopManagerPtr&
     linear_ir.insert_node(loop_end, loop_end_inputs, outer_loop_ids, false, loop_bounds.second);
 }
 
+bool InsertLoops::is_loop_dynamic(const LinearIR::LoopManager::LoopInfoPtr& loop_info) {
+    auto is_loop_port_dynamic = [](const LinearIR::LoopManager::LoopPort& port) {
+        return port.is_dynamic();
+    };
+    const auto& entry_points = loop_info->get_entry_points();
+    const auto& exit_points = loop_info->get_exit_points();
+    return utils::is_dynamic_value(loop_info->get_work_amount()) ||
+           std::any_of(entry_points.cbegin(), entry_points.cend(), is_loop_port_dynamic) ||
+           std::any_of(exit_points.cbegin(), exit_points.cend(), is_loop_port_dynamic);
+}
```

Comment:
Just an idea: maybe it is worth to move this function to `LoopInfo` methods? What do you think?

Reply:
Honestly, I thought about that. But I'm not sure that it's good idea since this `is_dynamic` definition is needed only here. I mean we use the check only here and in other places we will interested in only `is_dynamic_value(work_amount)` for example (no need to check ports).
But thanks for the idea anyway!